user-pii-data-updater/src/main/scala/org/sunbird/job/user/pii/updater/domain/Event.scala

```diff
@@ -22,8 +22,34 @@ class Event(eventMap: java.util.Map[String, Any], partition: Int, offset: Long)
 
   def objType: String = readOrDefault[String]("object.type", "")
 
+  def fromUserProfile: Map[String, AnyRef] = readOrDefault[Map[String, AnyRef]]("edata.fromUserProfile", Map.empty[String, AnyRef])
+
+  def toUserProfile: Map[String, AnyRef] = readOrDefault[Map[String, AnyRef]]("edata.toUserProfile", Map.empty[String, AnyRef])
+
+  def assetInformation: Map[String, AnyRef] = readOrDefault[Map[String, AnyRef]]("edata.assetInformation", Map.empty[String, AnyRef])
+
+  def fromUserId: String = readOrDefault[String]("edata.fromUserProfile.userId", "user")
+  def toUserId: String = readOrDefault[String]("edata.toUserProfile.userId", "user")
+
   def validEvent(): Boolean = {
-    validEventAction.contains(action) && StringUtils.isNotBlank(userId)
+    validEventAction.contains(action) && StringUtils.equalsIgnoreCase("User", objType) && (action match {
+      case "delete-user" => StringUtils.isNotBlank(userId)
+      case "ownership-transfer" => validateFromUserProfile(fromUserProfile) && validateToUserProfile(toUserProfile)
```

Comment:
Can we just use StringUtils.isNotBlank(fromUserId) && validateToUserProfile(toUserProfile) instead of having the method validateFromUserProfile ?

Reply:
the method is created, so that we can add more validation logic in future